benches/bench.rs

```diff
@@ -170,3 +170,51 @@ fn size_22_4m(b: &mut test::Bencher) {
 fn size_23_8m(b: &mut test::Bencher) {
     bench_size(b, 8 * 1024 * 1024);
 }
+
+macro_rules! gen_poll_test {
+    ($fnname:ident, $to_send:expr, $n:expr) => {
+        #[bench]
+        fn $fnname(b: &mut test::Bencher) -> () {
+            let mut active = Vec::with_capacity($to_send);
+            let mut dormant = Vec::with_capacity($n - $to_send);
+            let mut rx_set = IpcReceiverSet::new().unwrap();
+            let mut i = 0;
+            while i < $to_send {
+                let (tx, rx) = ipc::channel().unwrap();
+                rx_set.add(rx).unwrap();
+                active.push(tx);
+                i += 1;
+            }
+            while i < $n {
+                let (tx, rx) = ipc::channel::<String>().unwrap();
+                rx_set.add(rx).unwrap();
+                dormant.push(tx);
+                i += 1;
+            }
+            b.iter(|| {
+                for tx in active.iter() {
+                    tx.send("Just a flesh wound".to_owned()).unwrap();
+                }
+                let mut received = 0;
+                while received < $to_send {
+                    for result in rx_set.select().unwrap().into_iter() {
+                        let (_, received_data) = result.unwrap();
+                        let received_string: String = received_data.to().unwrap();
+                        assert_eq!(received_string, "Just a flesh wound");
+                        received += 1;
+                    }
+                }
+            });
+        }
+    }
+}
+
+#[cfg(test)]
+mod poll {
+    use ipc_channel::ipc::{self, IpcReceiverSet};
+    use test;
+
+    gen_poll_test!{bench_one_percent, 1, 100}
+    gen_poll_test!{bench_fifty_percent, 50, 100}
+    gen_poll_test!{bench_one_hundred_percent, 100, 100}
```

Comment:
I'm sure we could get more meaningful cases here: while the 100% one is useful to check the extreme end, the 50% one really doesn't tell us much. I'd rather go for 2 and 5 or something like that, which are way more likely in the real world I suspect...

Also, tests with smaller sets would be useful -- including the extreme (but not unlikely) case of just a single channel; and maybe 5 and 20 or so for likely real world cases.

(If you are *really* ambitious, you could add some instrumentation, to get an idea of the usage patterns actually coming up in Servo, so we don't have to rely on guesswork... I must admit though that I would only do that myself if I have absolutely no idea what might be realistic :-) )

On a related note, I guess it might be useful to also have some tests for the costs of creating/manipulating/destroying the set...